### DISCUSSION TITLE
How to apply to app on DenyList ?

### DISCUSSION THREAD

**Discussion by KaKi87** [xfqwdsj/IAmNotADeveloper#36] [Q&A] [answered]
Hello,

A contactless payment app refuses to work because of the device being rooted so I put it on DenyList, but now it refuses to work because of debugging being enabled, except LSPosed says applying IAmNotADeveloper on it wouldn't work because of being on DenyList.

What to do ?

Thanks

**Comment by xfqwdsj** [comment]
Try <https://github.com/LSPosed/LSPosed.github.io/releases> and only to disable deny list but not remove your app from it. This module follow Magisk's deny list and hide root to them.

But now I realized my module's working method can cause Zygisk to be found. I think I can hook the system and maintain its own app list to hide debugging mode.

**Reply by KaKi87** [reply]
That's precisely what I'm using :

![](https://media.discordapp.net/attachments/418482899034177557/1172881979218350140/Screenshot_20231111-135314_LSPosed.png)

And I did try it anyway, without success, even after rebooting.

Thanks

**Reply by xfqwdsj** [reply]
> Try https://github.com/LSPosed/LSPosed.github.io/releases

Have you checked it?

**Reply by KaKi87** [reply]
What is "Shamiko" ?

I can't find any source code, documentation, issue tracker or anything about it.

Thanks

**Reply by xfqwdsj** [reply]
From README:

### Introduction
Shamiko is a Zygisk module to hide Magisk root, Zygisk itself and Zygisk modules like riru hide.

Shamiko read the denylist from Magisk for simplicity but it requires denylist enforcement to be disabled first.

### Usage
1. Install Shamiko and enable Zygisk and reboot
2. Configure denylist to add processes for hiding
3. *DO NOT* turn on denylist enforcement

---

However, it does not provide source code. So if you don't use Shamiko and want to make the module work while hiding the root, it's impossible.

**Reply by xfqwdsj** [reply, marked as answer]
Magisk's deny list actually makes Zygisk not work for those apps, so once an app is added to the deny list nothing works (at least for notdeveloper, if I don't change the way it works one day).

**Comment by arifinnmo** [comment]
any news for this issue?

the zygisk is exposed when hiding developer mode using this app, so the target applications is not working
![Screenshot_2024-09-17-17-17-53-431-edit_io github vvb2060 mahoshojo](https://github.com/user-attachments/assets/3fd5c88f-7087-46f5-b33c-47189a96e396)

**Reply by xfqwdsj** [reply]
#4
